sdks/python/apache_beam/io/gcp/bigquery_file_loads.py

```diff
@@ -490,18 +512,39 @@ def __init__(
     # job to run - and thus we avoid using temporary tables
     self.temp_tables = True if callable(destination) else False
 
+    self._validate = validate
+    if self._validate:
+      self.verify()
+
+  def verify(self):
+    if (isinstance(self._custom_gcs_temp_location, str) and
+        not self._custom_gcs_temp_location.startswith('gs://')):
+      # Only fail if the custom location is provided, and it is not a GCS
+      # location.
+      raise ValueError('Invalid GCS location.\n'
+                       'Writing to BigQuery with FILE_LOADS method requires a '
+                       'GCS location to be provided to write files to be '
+                       'loaded into BigQuery. Please provide a GCS bucket, or '
+                       'pass method="STREAMING_INSERTS" to WriteToBigQuery.')
+
   def expand(self, pcoll):
     p = pcoll.pipeline
 
+    self._custom_gcs_temp_location = (
+        self._custom_gcs_temp_location
+        or p.options.view_as(GoogleCloudOptions).temp_location)
+
     load_job_name_pcv = pvalue.AsSingleton(
         p
         | "ImpulseJobName" >> beam.Create([None])
         | beam.Map(lambda _: _generate_load_job_name()))
 
     file_prefix_pcv = pvalue.AsSingleton(
         p
-        | "CreateFilePrefixView" >> beam.Create([self._input_gs_location])
-        | "GenerateFilePrefix" >> beam.Map(_generate_file_prefix))
+        | "CreateFilePrefixView" >> beam.Create(
+            [self._custom_gcs_temp_location])
```

Comment:
Do we want to support ValueProvider values of `custom_gcs_temp_location`?

Reply:
I've added a TODO. I'll add that in a later change.